desc/basis.py

```diff
@@ -317,6 +316,34 @@ def __repr__(self):
             )
         )
 
+    def __hash__(self):
+        """Get the hash of the object."""
+        return hash(
+            (
+                self.__class__.__name__,
+                self._L,
+                self._M,
+                self._N,
+                self._NFP,
+                self._sym,
+                self._spectral_indexing,
+            )
+        )
+
+    def __eq__(self, other):
+        """Check if two basis objects are equal."""
+        if not isinstance(other, _Basis):
+            return False
+        return (
```

Comment:
usually the first condition checks reference equality e.g. `self is other` to short-circuit